### ISSUE TITLE
[critical] EOS is never supervised when `pad_token` falls back to `eos_token` — model never learns to stop

### ISSUE THREAD

**Issue by nagohachi** [nagohachi/glide#7]
## Defect

`ensure_special_tokens` silently sets `tokenizer.pad_token = tokenizer.eos_token` for any base model lacking a pad token ([models/special_tokens.py#L62-L63](https://github.com/nagohachi/glide/blob/1cacf15da0d5e976c3db81648c829ee15535ab1a/src/glide/models/special_tokens.py#L62-L63)) — that's Llama, Gemma, Mistral, i.e. the advertised composed bases.

Both collators then mask **all** pad positions out of the loss:

- [data/composed_collator.py#L148-L151](https://github.com/nagohachi/glide/blob/1cacf15da0d5e976c3db81648c829ee15535ab1a/src/glide/data/composed_collator.py#L148-L151)
- [data/collator.py#L140-L142](https://github.com/nagohachi/glide/blob/1cacf15da0d5e976c3db81648c829ee15535ab1a/src/glide/data/collator.py#L140-L142)

```python
labels[input_ids == pad_id] = -100
```

When `pad == eos`, this also erases the genuine response-terminating EOS from the labels. The model is never trained to emit EOS, so validation/test generations run to `max_new_tokens` on every sample (which also silently inflates decode time and can hurt WER via trailing babble).

## Fix

Mask by `attention_mask == 0` (position-based) instead of by token id, or keep the first EOS after the prompt unmasked. Both collators need the fix. Independently flagged by two audit passes (data + models).
